### DISCUSSION TITLE
Random "TypeError: Load failed" response for calls to Supabase every once in a while. Why might this happen?

### DISCUSSION THREAD

**Discussion by isaacschiffler** [supabase/supabase#27477] [Questions]
In my React app, I am currently facing an issue where randomly, every once in a while, a call returns an error to the website instead of the data from the call to Supabase. I am having a difficult time debugging it. Supabase shows that it is not failing on its side. Every time it happens I am logging the error, but I still can't really figure out a reason for the bug. My app is also deployed as a static web app with Azure if that helps at all. It seems to happen randomly for all my API calls. It is not quite evenly distributed amongst them. Any and all help would be much appreciated. It's very weird that the API calls *work most of the time*, but produce an error every once in a while. If the proportion of errors was lower I would maybe ignore but it seems like I get about 80 or so errors a day for about 1200 people using the site in the same day. Also, I am on the free Supabase plan if that might factor in. I'll put my compute usage at the bottom.

Here is a code snippet for one of the calls (they are all quite similar but some call Supabase functions instead):

      const { data, error } = await supabase
        .from('all_questions')
        .select('*')
        .eq('day', formattedDate);

      if (error) {
        console.error('Error fetching data:', error.message);
        report_error(userID, "read current question", error.message, (Date.now() - sessionStart) / 1000, String(formattedDate), JSON.stringify(data));
        throw error;
      }

      setTodaysData(data);
      console.log("Data fetched successfully!");

`report_error()` is a function that inserts an error log into supabase so I can track them)

Here is the information I log when an error occurs:

![my error logging](https://i.sstatic.net/TUUDFtJj.png)

Error distribution:

![error distribution](https://i.sstatic.net/c9MsOWgY.png)

Here is Supabase error tracking showing no errors:

![supabase error log](https://i.sstatic.net/xFwvO5Yi.png)

Supabase compute usage:

Memory: 45%  
Swap: 20%  
Average CPU: .6%  
Max CPU: 1.76%  
Disk IO: 1%

I have tried researching what the error message means but I can't find too much on it specifically. Can't think of what might cause it. I expect the API call to Supabase to return data, but it is producing an error randomly.

**Comment by silentworks** [comment]
Do you get this error when running the site locally? it might be worth commenting out functionality line by line to see if the error is caused by one functionality or only when many are being run. I've never seen this error message before. Is this happening in a specific browser as someone in another issue stated they had this happen to them in Safari https://github.com/supabase/supabase/issues/20982

**Reply by isaacschiffler** [reply]
I have not seen it happen when running locally. I can try out your idea to comment out functionality line by line. The most confusing part about this is the fact that it randomly happens to all my endpoints, not just one specific endpoint. I don't believe this is happening in a specific browser. I couldn't find too many parallels between the discussion you sent and my problems.

**Reply by encima** [reply]
Can you confirm that this is not happening in a specific browser? Possibly logging the User Agent with the error.

**Reply by isaacschiffler** [reply]
yes, adding this error log now and will come back after seeing results.

**Reply by isaacschiffler** [reply]
It does look like all errors are stemming from Safari. Doesn't look like the issue linked above had any solutions. Any further ideas?

**Reply by Yassdrk** [reply]
Hello,
Got the same issue since today. 
user agent: 
Mozilla/5.0 (iPhone; CPU iPhone OS 17_5_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Mobile/15E148 Safari/604.1

**Comment by Raj1v** [comment]
Any updates? We are facing the exact same issue, also coming predominatly from Safari.

**Reply by isaacschiffler** [reply]
It never got fixed for me... I ended up adding an API layer web app for the calls to run through so I am no longer calling supabase from the client, aka no calls directly from Safari. This resolved it for me but just annoying to have to do just for this bug.

**Reply by Raj1v** [reply]
Thanks for sharing. We have patched it by overwriting the `fetch` function that the Supabase JS SDK uses to retry any call three times before failing.

**Reply by bahoan** [reply]
I'm facing the same issue and I've also noticed that it only happens with clients using Safari. For example, I had this user with the following user agent:

Mozilla/5.0 (iPhone; CPU iPhone OS 17_7 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.7 Mobile/15E148 Safari/604.1

Has anyone managed to find a solution while still calling Supabase from the client?

**Comment by scoover** [comment]
My app tracks a sporting event with about 500 database writes per 90-minute game. We are typically seeing this error occur about 2-5 times per game with error message: `Load failed`

A few observations:
- There is a pattern to the failures that suggests the error frequency increases when the smartphone has been idle and then is abruptly invoked for database writes. Is it possible others see the failures connected to "wake-up" type usage?
- Despite many attempts, I have not been able to reproduce the error on a laptop at home. All errors that I'm aware of are on a smartphone.
- I have also struggled to reproduce this issue _at home_ even on my smartphone. My suspicion is that it is related to cell service quality out on the fields— e.g.  added latency? race condition?
- Many users are on mobile Safari but the error occurs on non-Safari platforms as well.
- I have code that flags the error to the user. Most users say they can run the same command immediately after the modal with no issue... so the issue appears to be transient and not baked into core code implementation.

Really eager to get this fixed as it has frustrated many users. 
Glad to see I'm not alone on this. 🙃

**Reply by Raj1v** [reply]
We have the exact same observations on the last 4 points

**Reply by zxqp** [reply]
Same here

**Comment by Raj1v** [comment]
Also for Auth, some users are facing Load Failed when signing in. User shared video of using safari on iOS, but I could not replicate the behavior myself on iOS. @encima Any chance the Supabase team knows more on this issue?

**Comment by GregoryKogan** [comment]
I figured that this error comes up much more often when I use Supabase dashboard in Safari. I have to open the dashboard in another browser like Firefox while developing frontend in Safari. The error still appears once in a while, but quitting and reopening Safari usually helps.

**Comment by IPJT** [comment]
I have the same issue! 
Did anyone figure this out?

**Comment by chanon** [comment]
Everyone, please see my comment here:
https://github.com/supabase/supabase/issues/20982#issuecomment-2944288493

I am quite confident that is the cause for this issue.
